### PR TITLE
[dev v2.7] k8s december patches

### PR DESCRIPTION
**Pull request by vardhaman22**
related issue: https://github.com/rancher/rancher/issues/39755

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -7846,6 +7846,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			CertDownloader:            "rancher/rke-tools:v0.1.88",
+			KubernetesServicesSidecar: "rancher/rke-tools:v0.1.88",
+			KubeDNS:                   "rancher/mirrored-k8s-dns-kube-dns:1.17.4",
+			DNSmasq:                   "rancher/mirrored-k8s-dns-dnsmasq-nanny:1.17.4",
+			KubeDNSSidecar:            "rancher/mirrored-k8s-dns-sidecar:1.17.4",
+			KubeDNSAutoscaler:         "rancher/mirrored-cluster-proportional-autoscaler:1.8.3",
+			Flannel:                   "rancher/mirrored-coreos-flannel:v0.15.1",
+			FlannelCNI:                "rancher/flannel-cni:v0.3.0-rancher6",
+			CalicoNode:                "rancher/mirrored-calico-node:v3.21.1",
+			CalicoCNI:                 "rancher/calico-cni:v3.21.3-rancher1",
```
Comment (kinarashah) [inline rke/k8s_rke_system_images.go]: afaik, 1.22.17 also would use the new `v3.22.5-rancher1-rc1`

Comment (snasovich) [reply rke/k8s_rke_system_images.go]: Yup, good catch. Also, https://github.com/rancher/kontainer-driver-metadata/pull/1030#discussion_r1042775677

Comment (vardhaman22) [reply rke/k8s_rke_system_images.go]: done

**Review by snasovich — COMMENTED** [review]
Changes themselves LGTM, waiting on https://github.com/rancher/kontainer-driver-metadata/pull/1030

Now, we do currently reference `rancher/calico-cni:v3.22.5-rancher1-rc1`.
@vardhaman22 @mitulshah-suse @rayandas , once the PR above is updated and merged and this PR is rebased, can we do some preliminary testing on this source fork to confirm it's working OK? Then I can tag `rancher/calico-cni:v3.22.5-rancher1` and we update this PR so non-RC version reference get merged.

We can also merge with RC and do testing and then do another PR to update from RC1 to non-RC. Let me know what option works better for you.

**Comment by rayandas** [comment]
Sure @snasovich, merging this PR with the RC tag would be a good idea so that QAs can test and confirm if everything is working fine or not. After that we can just raise another KDM PR with un-RC tag.
cc: @vardhaman22

**Review by snasovich — APPROVED** [review]
LGTM, the newly introduced versions reference new Calico/Canal images and Hyperkube as expected.

Wait for https://github.com/rancher/kontainer-driver-metadata/pull/1030 to be merged first.

**Review by kinarashah — APPROVED** [review]
For future, we could also consider tagging `-rancher1` right away and tagging another version if further fixes are required. Would save another cycle of PRs.

**Pull request merged**